File: habit_backend/models/mission.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/database");
const User = require("./user");

const Mission = sequelize.define("mission", {
  title: { type: DataTypes.STRING, allowNull: false },
  description: { type: DataTypes.STRING },
  type: { type: DataTypes.STRING, allowNull: false },
  is_completed: { type: DataTypes.BOOLEAN, defaultValue: false },
});

Mission.belongsTo(User, { foreignKey: "user_id" });
User.hasMany(Mission, { foreignKey: "user_id" });

module.exports = Mission;

File: front_flutter/lib/models/mission.dart
class Mission {
  final int id;
  final String title;
  final String description;
  final String type;
  final bool isCompleted;

  Mission({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.isCompleted,
  });

  factory Mission.fromJson(Map<String, dynamic> json) {
    return Mission(
      id: json['id'],
      title: json['title'],
      description: json['description'] ?? '',
      type: json['type'],
      isCompleted: json['is_completed'],
    );
  }
}
